app/src/androidTestWebview/java/org/mozilla/focus/activity/SearchEngineSelectionTest.java

```diff
@@ -60,85 +74,140 @@ public void tearDown() throws Exception {
-        assertEquals(TestHelper.inlineAutocompleteEditText.getText(), "mozilla focus");
-        TestHelper.pressEnterKey();
-        googleWebView.waitForExists(waitingTime);
-        TestHelper.progressBar.waitForExists(webPageLoadwaitingTime);
-        Assert.assertTrue(TestHelper.progressBar.waitUntilGone(webPageLoadwaitingTime));
-        assertTrue (TestHelper.browserURLbar.getText().contains("google"));
-        assertTrue (TestHelper.browserURLbar.getText().contains("mozilla"));
-        assertTrue (TestHelper.browserURLbar.getText().contains("focus"));
-
-         /* Now do another search */
-        TestHelper.browserURLbar.click();
-        TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
-        TestHelper.inlineAutocompleteEditText.clearTextField();
-        TestHelper.inlineAutocompleteEditText.setText("mozilla focus");
-        TestHelper.hint.waitForExists(waitingTime);
-
-        // Check the search hint bar is correctly displayed
-        assertTrue(TestHelper.hint.getText().equals("Search for mozilla focus"));
-        TestHelper.hint.click();
+
+        // Open [settings menu] and select Search
+        {
+            assertTrue(TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime));
+            openSettings();
+            searchEngineSelectorLabel.click();
+        }
+
+
+        // Open [settings menu] > [search engine menu] and click "Default" label
+        // note: we don't click on default search engine itself cause this will change
+        {
+            UiObject defaultSearchEngineLabel = TestHelper.mDevice.findObject(new UiSelector()
+                    .text("Default")
+                    .resourceId("android:id/summary"));
+
+            defaultSearchEngineLabel.click();
+        }
+
+
+        // Open [settings menu] > [search engine menu] > [search engine list menu]
+        // then select desired search engine
+        {
+            UiScrollable searchEngineList = new UiScrollable(new UiSelector()
+                    .resourceId(TestHelper.getAppName() + ":id/search_engine_group").enabled(true));
+
+            UiObject defaultEngineSelection = searchEngineList.getChildByText(new UiSelector()
+                    .className(RadioButton.class), mSearchEngine);
+
+            defaultEngineSelection.waitForExists(waitingTime);
+            assertTrue(defaultEngineSelection.getText().equals(mSearchEngine));
+            defaultEngineSelection.click();
+        }
+
+
+        // Verify new engine default
+        {
+            TestHelper.pressBackKey();
+            TestHelper.settingsHeading.waitForExists(waitingTime);
+
+            UiObject defaultSearchEngine = TestHelper.mDevice.findObject(new UiSelector()
+                    .text(mSearchEngine)
+                    .resourceId("android:id/title"));
+
+            assertTrue(defaultSearchEngine.getText().equals(mSearchEngine));
+            TestHelper.pressBackKey();
+            TestHelper.pressBackKey();
+        }
+
+
```

Comment:
let's only use single empty line between a chunk of actions, instead of 2

Reply:
k